src/mnemonics/croatian.h

```diff
@@ -0,0 +1,1686 @@
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+/*!
+ * \file croatian.h
+ *
+ * \brief Croatian word list and map.
+ */
+
+#ifndef CROATIAN_H
+#define CROATIAN_H
+
+#include <vector>
+#include <unordered_map>
+#include "language_base.h"
+#include <string>
+
+/*!
+ * \namespace Language
+ * \brief Mnemonic language related namespace.
+ */
+namespace Language
+{
+  class Croatian: public Base
+  {
+  public:
+    Croatian(): Base("Hrvatski", std::vector<std::string>({
+        "abeceda",
+        "adaptacija",
+        "admiral",
+        "adresa",
+        "advokat",
+        "aerodrom",
+        "afera",
+        "afganistan",
+        "afrika",
+        "agent",
+        "agonija",
+        "agresija",
+        "akademija",
+        "akcija",
+        "aktivirati",
+        "akumulator",
+        "akvarij",
+        "alarm",
+        "algoritam",
+        "alkohol",
+        "alternativa",
+        "aluminij",
+        "aljaska",
+        "ambasador",
+        "amerika",
+        "amputirati",
+        "amsterdam",
+        "analizirati",
+        "anđeo",
+        "anestetik",
+        "anonimno",
+        "apartman",
+        "apetit",
+        "aplauz",
+        "apsolutno",
+        "arena",
+        "argument",
+        "arhitekt",
+        "arogantan",
+        "arsenal",
+        "asistent",
+        "asocijacija",
+        "aspirin",
+        "astronaut",
+        "atentat",
+        "atmosfera",
+        "atomske",
+        "atrakcija",
+        "audicija",
+        "aukcija",
+        "aurora",
+        "automobil",
+        "avantura",
+        "avenija",
+        "avion",
+        "azija",
+        "babaroga",
+        "baciti",
+        "bačena",
+        "badnjak",
+        "bagdad",
+        "bajka",
+        "baka",
+        "bala",
+        "bambi",
+        "bandit",
+        "barem",
+        "baterija",
+        "bauer",
+        "baviti",
+        "baza",
+        "beba",
+        "beč",
+        "bedro",
+        "bejzbol",
+        "berlin",
+        "besplatno",
+        "bezveze",
+        "biblija",
+        "bicikl",
+        "bičevati",
+        "biće",
+        "biftek",
+        "bijel",
+        "bik",
```

Comment:
Three unit tests fail because "bik" is too short.

`Error initializing mnemonics: Too short word in Hrvatski word list: bik`

```
[  FAILED  ] mnemonics.consistency
[  FAILED  ] mnemonics.all_languages
[  FAILED  ] mnemonics.language_detection_with_bad_checksum
```
Plus, there are same words with and without croatian diacritics in the list.

Reply:
Can you try after setting 3 for prefix length (see previous comment) ?

Reply:
Previously failing mnemonics unit tests pass when prefix is set to 3.

Reply:
Thanks. I'll OK this once the s/4/3/ is fixed and the two patches squashed then.

Reply:
@iDunk5400 "Plus, there are same words with and without croatian diacritics in the list." <- not sure what's your reasoning here. Forcing diacritics in all words doesn't make any sense. It is not that we in Croatian use `šđčćž` in all words. Though, it would be quite funny to force it in official grammar.

Reply:
@moneromooo-monero will "patch" and "squash" tomorrow as my cloned repo is at the other computer. Just one question. Is there any way you prefer "squashing" commits in pull requests?

Reply:
@moneromooo-monero squashed, "s/4/3/" and pushed

Reply:
I'm talking about citati and čitati, cujem and čujem etc...

Reply:
@iDunk5400 you are right here. need to recheck those. thank you for pointing it out. @moneromooo-monero please let me double check this kind of cases @iDunk5400 mentions. will commit and squash changes

Reply:
I don't quite understand "Is there any way you prefer "squashing" commits in pull requests?". I much prefer patches to previous commits to be in the original commit (ie, don't introduce a bug just to remove it in the same patch set). Does this answer the question ?

Reply:
@moneromooo-monero fully understand

Reply:
@iDunk5400 done. Changed "citati" to "citat" as it is a valid word (in Croatian for "quotation"). Changed "cujem" to "coflek" (totally different meaning). Double checked everything else (created a mini script to translate diacritics to latin and see if there are more "conflicts"). Committed and pushed changes